app/containers/Home/CreateRoom.js

```diff
@@ -1,24 +1,28 @@
-import React, { useState } from 'react';
-
-import { setUserName } from 'actions/root';
-
+import React, { useState, useEffect } from 'react';
 import { css } from 'emotion';
-import { Container, Col, Row, Input, Button, Form } from 'reactstrap';
 import { useHistory } from 'react-router-dom';
+import { Container, Col, Row, Input, Button, Form } from 'reactstrap';
 import { useDispatch, useSelector } from 'react-redux';
+import { createRoomAction } from '../../redux/actions/room';
+import Spinner from '../../components/Spinner/Spinner';
 
 const CreateRoom = () => {
   const dispatch = useDispatch();
   const rootState = useSelector((state) => state);
+  const history = useHistory();
+
+  const [activeRoom, setActiveRoom] = useState(rootState.room.active);
+  useEffect(() => {
```

Comment:
Can you explain why useEffect() is needed here? I looked up the useEffect hook in the official react docs and can't wrap my head around what it's actually doing and why it's needed.